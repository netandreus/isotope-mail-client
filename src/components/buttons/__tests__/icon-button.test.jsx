import React from 'react';
import {shallow} from 'enzyme/build/index';
import IconButton from '../icon-button';

describe('IconButton component test suite', () => {
  test('Snapshot render, should render icon-button', () => {
    // Given
    const props = {className: 'higher-class', onClick: jest.fn(), disabled: false};
    // When
    const iconButton = shallow(<IconButton {...props}>material_icon_name</IconButton>);
    // Then
    expect(iconButton).toMatchSnapshot();
  });
  test('click, should trigger function', () => {
    // Given
    const props = {className: 'higher-class', onClick: jest.fn(), disabled: false};
    const iconButton = shallow(<IconButton {...props}>material_icon_name</IconButton>);
    // When
    iconButton.find('button').simulate('click');
    // Then
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
