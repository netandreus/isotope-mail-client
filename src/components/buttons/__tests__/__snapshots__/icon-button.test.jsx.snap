// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component test suite Snapshot render, should render icon-button 1`] = `
<button
  className="material-icons mdc-icon-button higher-class"
  disabled={false}
  onClick={[MockFunction]}
>
  material_icon_name
</button>
`;
